File: Add Binary/Add Binary.swift
import Foundation

/*
 Given two binary strings, return their sum (also a binary string).
 */
class Solution {
    func addBinary(_ a: String, _ b: String) -> String {
        var result = ""
        var sum = 0
        let aChars = Array(a.characters)
        let bChars = Array(b.characters)
        var i = aChars.count - 1
        var j = bChars.count - 1
        var evl = 0

        while i>=0 || j >= 0 || evl > 0 {
            sum = evl // 进1位
            if i>=0 {
                sum += Int(String(aChars[i]))!
                i -= 1
            }
            if j>=0 {
                sum += Int(String(bChars[j]))!
                j -= 1
            }
            // sum 0, 1, 2
            // evl 0, 1
            evl = sum / 2
            sum = sum % 2
            result = String(sum) + result
        }

        return result
    }
}

let a = Solution()
let b = a.addBinary("11", "10")

File: Add Binary/Add Binary.py
#! /usr/bin/env python
# -*- coding: utf-8 -*-

class Solution(object):
    def addBinary(self, a, b):
        """
        :type a: str
        :type b: str
        :rtype: str
        """
        result = ""
        sumVal = 0
        achars = list(a)
        bchars = list(b)
        i = len(achars) - 1
        j = len(bchars) - 1
        evl = 0

        while i >= 0 or j >= 0 or evl>0:
            sumVal = evl
            if i >= 0:
                sumVal += int(achars[i])
                i -= 1
            if j >= 0:
                sumVal += int(bchars[j])
                j -= 1

            evl = sumVal // 2
            sumVal = sumVal % 2
            result = str(sumVal) + result

        return result


a = Solution()
b = a.addBinary('1','1')
print(b)